lib/tramway/helpers/decorate_helper.rb

```diff
@@ -0,0 +1,29 @@
+# frozen_string_literal: true
+
+module Tramway
+  module Helpers
+    # Provides methods into Rails ActionController
+    #
+    module DecorateHelper
+      def tramway_decorate(object_or_array, decorator: nil)
+        decorator_class(object_or_array, decorator).decorate object_or_array, self
+      end
+
+      private
+
+      def decorator_class(object_or_array, decorator)
+        if decorator.present?
+          decorator
+        else
+          klass = if object_or_array.class.name.in? ['ActiveRecord::Relation', 'Array']
+                    object_or_array.first.class
+                  else
+                    object_or_array.class
+                  end
```

Comment:
```rb
object = object_or_array.first if object_or_array.class.name.in? ['ActiveRecord::Relation', 'Array']
object ||= object_or_array
object.class
```

BTW there is a duplicate of `*.class.name.in? ['ActiveRecord::Relation', 'Array']` I think we can move it to the helper

Reply:
```rb
object = object_or_array.first if object_or_array.class.name.in? ['ActiveRecord::Relation', 'Array']
object ||= object_or_array

"#{object.class}Decorator".constantize
```

Reply:
Fixed!